content/posts/2025-02-26-whats_new_1_8.md

```diff
@@ -0,0 +1,295 @@
+
+```
+spec:
+  featureSet: CustomNoUpgrade
+  customNoUpgrade:
+    enabled:
+      - OVNObservability
+```
+
+<p style="text-align: center">Listing 3: Enable OVNObservability feature gate</p>
+
+This can take upwards of *10+ minutes* for this to take effect, so be patient.  Initially, it might seem like nothing has happened, but after about five minutes, you will lose connection to your cluster because OVN-Kubernetes has to restart.  Then after another five minutes, everything should be back online.
+
+To enable this feature in FlowCollector, enter `oc edit flowcollector` and configure the following in the **ebpf** section:
+
+```
+spec:
+  agent:
+    ebpf:
+      sampling: 1  # recommended so all flows are observed
+      privileged: true
+      features:
+      - NetworkEvents
+```
+
+<p style="text-align: center">Listing 4: Enable NetworkEvents feature in FlowCollector</p>
+
+A packet could be dropped for many reasons, but often times, it's due to a network policy.  In OCP, there are three different types of network policies.  Network Events supports all three.  Click the links to learn how to configure them.
+
+1. [Kubernetes Network Policy](https://docs.openshift.com/container-platform/4.18/networking/network_security/network_policy/about-network-policy.html) - similar to a firewall rule, typically based on an IP address, port, and protocol
+2. [Admin Network Policy](https://docs.openshift.com/container-platform/4.18/networking/network_security/AdminNetworkPolicy/ovn-k-anp.html) - allows cluster admin to create policies that are at the cluster level
+3. [Egress Firewall](https://docs.openshift.com/container-platform/4.18/networking/network_security/egress_firewall/viewing-egress-firewall-ovn.html) - controls what external hosts a pod can connect to
+
+In Figure 3, the Traffic flows table shows a column for Network Events.  The first row shows a drop from a network policy and the second row allowed the packet in an admin network policy.
+
+![Flows table with Network Events]({page.image('whats-new-1-8/flows-network_events.png')})
+Figure 3: Flows table with Network Events
+
+### eBPF Flow Filter Enhancements
+
+eBPF flow filter was first introduced in Network Observability 1.6.  It lets you decide what you want to observe at the initial collector stage, so you can minimize the resources used by observability.  It comes with more enhancements in this release.
+
+#### More than one flow filter rule
+
+Instead of one filter rule, you can now have up to 16 rules, thus removing a major limitation of this feature.  The order of preference is based on the most specific match, which is the longest prefix match of the CIDR or if tied, the longest prefix match of the peer CIDR.  For example, if one rule has a CIDR of 10.0.0.0/16 and another with 10.0.1.0/24, then if the address is 10.0.1.128, it would match the second rule because that's more specific (24-bit prefix vs. 16-bit prefix).  This also implies no two rules can have the same CIDRs.
```

Comment:
I think it's important to understand that only one rule gets evaluated per flow IP, ie. if a rule is selected (based on CIDR & prefix precedence) but doesn't match the other criteria, it's not going to run the "next" rule for that same IP. (Exactly what we said yesterday in the meeting)

Up to you to call that out or not - I thought it was explicitly said in @msherif1234 's blog, but it's actually not very explicit, the algorithm is described but without emphasize on that point.

Reply:
Reworded